homeassistant/components/group/notify.py

```diff
@@ -48,19 +60,108 @@ def update(input_dict, update_source):
 
 async def async_get_service(hass, config, discovery_info=None):
     """Get the Group notification service."""
-    return GroupNotifyPlatform(hass, config.get(CONF_SERVICES))
+    return GroupNotifyPlatform(
+        hass,
+        config.get(CONF_NAME),
+        config.get(CONF_SERVICES),
+        config.get(CONF_SWITCH),
+        config.get(CONF_MAX_PER_MINUTE),
+        config.get(CONF_MAX_PER_HOUR),
+        config.get(CONF_MAX_PER_DAY),
+    )
 
 
 class GroupNotifyPlatform(BaseNotificationService):
     """Implement the notification service for the group notify platform."""
 
-    def __init__(self, hass, entities):
+    def __init__(
+        self, hass, name, entities, switch, max_per_minute, max_per_hour, max_per_day
+    ):
         """Initialize the service."""
         self.hass = hass
+        self.name = name
         self.entities = entities
+        self.switch = switch
+        self.max_per_minute = max_per_minute
+        self.max_per_hour = max_per_hour
+        self.max_per_day = max_per_day
+        self.lock = None
+        if self.max_per_minute or self.max_per_hour or self.max_per_day:
+            # we only push onto the deques that have rate limits
+            self.calls_minute = deque()
+            self.calls_hour = deque()
+            self.calls_day = deque()
+            self.lock = threading.Lock()
```

Comment:
We can't use a threading lock in async context. Only asyncio locks are allowed.

Reply:
Ah, good catch... will fix to use asyncio.lock.  Thx!